src/components/legend/draw.js

```diff
@@ -452,43 +452,127 @@ function computeLegendDimensions(gd, groups, traces) {
         opts = fullLayout.legend,
         borderwidth = opts.borderwidth;
 
-    if(helpers.isGrouped(opts)) {
-        groups.attr('transform', function(d, i) {
-            return 'translate(0,' + i * opts.tracegroupgap + ')';
+    if(helpers.isVertical(opts)) {
+        if(helpers.isGrouped(opts)) {
```

Comment:
It would be worthwhile to set  `var isGrouped = helpers.isGrouped(opts);` :arrow_double_up: